File: src/main/java/cn/dsnbo/bedrockplayersupport/listener/login/AuthMeListener.java
package cn.dsnbo.bedrockplayersupport.listener.login;

import cn.dsnbo.bedrockplayersupport.BedrockPlayerSupport;
import cn.dsnbo.bedrockplayersupport.config.Config;
import cn.dsnbo.bedrockplayersupport.config.Language;
import cn.dsnbo.bedrockplayersupport.util.StringUtil;
import fr.xephi.authme.api.v3.AuthMeApi;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.geysermc.floodgate.api.FloodgateApi;

import java.util.UUID;

/**
 * @author DongShaoNB
 */
public class AuthMeListener implements Listener {

  @EventHandler
  public void onPlayerJoinEvent(PlayerJoinEvent event) {
    Player player = event.getPlayer();
    UUID playerUuid = player.getUniqueId();
    String playerName = player.getName();
    Config config = BedrockPlayerSupport.getMainConfigManager().getConfigData();
    Language language = BedrockPlayerSupport.getLanguageConfigManager().getConfigData();
    if (FloodgateApi.getInstance().isFloodgatePlayer(playerUuid)) {
      if (!AuthMeApi.getInstance().isAuthenticated(player)) {
        if (AuthMeApi.getInstance().isRegistered(playerName)) {
          if (config.enableLogin()) {
            AuthMeApi.getInstance().forceLogin(player);
            player.sendMessage(
                BedrockPlayerSupport.getMiniMessage().deserialize(language.loginSuccessfully()));
          }
        } else {
          if (config.enableRegister()) {
            String password = StringUtil.randomString(config.passwordLength());
            AuthMeApi.getInstance().registerPlayer(playerName, password);
            player.sendMessage(
                BedrockPlayerSupport.getMiniMessage().deserialize(language.registerSuccessfully()
                    .replaceAll("%password%", password)));
          }
        }
      }
    }
  }
}

File: src/main/kotlin/cc/dsnb/bedrockplayersupport/listener/auth/AuthMeListener.kt
package cc.dsnb.bedrockplayersupport.listener.auth

import cc.dsnb.bedrockplayersupport.BedrockPlayerSupport
import cc.dsnb.bedrockplayersupport.util.StringUtil
import fr.xephi.authme.api.v3.AuthMeApi
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

/**
 * @author DongShaoNB
 */
class AuthMeListener : Listener {

    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        val player = event.player
        val playerUuid = player.uniqueId
        val playerName = player.name
        val mainConfig = BedrockPlayerSupport.mainConfigManager.getConfigData()
        val langConfig = BedrockPlayerSupport.langConfigManager.getConfigData()
        BedrockPlayerSupport.scheduler.runTaskLater(player, {
            if (BedrockPlayerSupport.floodgateApi.isFloodgatePlayer(playerUuid)) {
                if (AuthMeApi.getInstance().isAuthenticated(player).not()) {
                    if (AuthMeApi.getInstance().isRegistered(playerName)) {
                        if (mainConfig.enableLogin()) {
                            AuthMeApi.getInstance().forceLogin(player)
                            player.sendMessage(StringUtil.formatTextToComponent(player, langConfig.loginSuccessfully()))
                        }
                    } else {
                        if (mainConfig.enableRegister()) {
                            val password = StringUtil.randomString(mainConfig.passwordLength())
                            AuthMeApi.getInstance().forceRegister(player, password)
                            player.sendMessage(
                                StringUtil.formatTextToComponent(
                                    player,
                                    langConfig.registerSuccessfully().replace("%password%", password)
                                )
                            )
                        }
                    }
                }
            }
        }, 20L)
    }

}